File: Baekjoon/5618.java
import java.io.*;
import java.util.ArrayList;
import java.util.StringTokenizer;

class Main {
    static int n;
    static ArrayList<Integer> num = new ArrayList<>();

    public static int gcd(int a, int b) {
        if (a % b == 0) return b;
        return gcd(b, a % b);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;

        n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < n; i++) num.add(Integer.parseInt(st.nextToken()));

        int g = gcd(num.get(0), num.get(1));
        if (n == 3) g = gcd(g, num.get(2));

        for (int i = 1; i <= g; i++)
            if (g % i == 0)
                bw.write(i + "\n");
        bw.flush();
    }
}

File: Baekjoon/5618.kt
import java.util.*

fun main() {
    fun gcd(a: Int, b: Int): Int = if (a % b == 0) b else gcd(b, a % b)

    val n = readLine()!!.toInt()
    val num = mutableListOf<Int>()
    val st: StringTokenizer = StringTokenizer(readLine())

    for (i in 1..n)
        num.add(st.nextToken().toInt())

    var g = gcd(num[0], num[1])
    if (n == 3) g = gcd(g, num[2])

    for (i in 1..g)
        if (g % i == 0)
            println(i)
}
